qiskit_rigetti/_qcs_job.py

```diff
@@ -107,6 +109,28 @@ def _start_circuit(self, circuit: QuantumCircuit) -> Response:
         # typing: QuantumComputer's inner QAM is generic, so we set the expected type here
         return cast(Response, self._qc.qam.execute(executable))
 
+    def _handle_barriers(self, qasm: str) -> str:
+        lines = []
+        for line in qasm.splitlines():
+            if not line.startswith("barrier"):
+                lines.append(line)
+                continue
+
+            qubits = line[(len("barrier ")) :]
+            num_qubits = qubits.count(",") + 1
+            if num_qubits != self._configuration.num_qubits:
+                warnings.warn(
+                    "barriers on qubit subsets will be omitted during execution on a RigettiQCSBackend -- "
+                    "apply barrier to all qubits to preserve gate operators",
+                )
+                continue
+
```

Comment:
I'm debating whether or not to warn here too, to let the user know that we replaced the barrier with an empty preserve block. Thoughts?

Reply:
nitpick: technically should say "proper subsets" because the set of all qubits is still a subset

Reply:
also maybe change second part to `apply barrier to all circuit qubits to preserve barrier effect`